guides/source/active_record_querying.md

```diff
@@ -24,32 +24,101 @@ Code examples throughout this guide will refer to one or more of the following m
 
 TIP: All of the following models use `id` as the primary key, unless specified otherwise.
 
+
 ```ruby
-class Client < ApplicationRecord
-  has_one :address
-  has_many :orders
-  has_and_belongs_to_many :roles
+class Author < ApplicationRecord
+  has_many :books, -> { order('year_published DESC') }
 end
 ```
 
 ```ruby
-class Address < ApplicationRecord
-  belongs_to :client
+class Book < ApplicationRecord
+
+  belongs_to :supplier
+  belongs_to :author, dependent: :destroy
+
+  has_many :reviews
+  has_and_belongs_to_many :orders, join_table: 'books_orders'
+ 
+  # default_scope { where('year_published >= ?', Date.current.year - 50 )}
+
+  scope :in_print, -> { where(out_of_print: false) }
+  scope :out_of_print, -> { where(out_of_print: true) }
+
+  scope :recent, -> { where('year_published >= ?', Date.current.year - 50 )}
+  scope :old, -> { where('year_published < ?', Date.current.year - 50 )}
+
+  scope :out_of_print_and_expensive, -> { out_of_print.where('price > 500') }
+
+  scope :costs_more_than, ->(amount) { where('price > ?', amount) }
```

Comment:
Good to make me check again.  I found 2 to remove.

 Here are the ones that are in use:
`in_print` : used in  Scopes - Merging of scopes
`out_of_print` : used in Scopes, Scopes - Merging of scopes, Removing All Scoping
`old` : used in Scopes - Merging of scopes
`out_of_print_and_expensive` : used in Scopes
`costs_more_than` : used in Scopes - Passing in arguments
`created_before` : used in Scopes - Using conditionals


removed:
`Book.recent`
`Customer.frequent_buyers`

Reply:
IMO it might be a better idea to hide the source to these models (either in a linked repository or somewhere out of sight initially) and make it easy for people to find. We don't show the source of most of the current pseudocode, but I like the idea of making it available.